CenterForOpenScience/osf.io
# [PLAT-846] Add related counts to relationship fields

## Purpose

The ember teams would like to know how many related items they are looking at before the request the API for more info, but certain related fields don't have counts associated with them. This PR adds that.

## Note

I considered writing tests for this be the ticket says this behavior will be generalized to apply to all relationship fields so I think tests should be held off until then, when we won't have to special case.

## Changes

- add related counts for user serializer for:
   - node
   - quickfiles
   - institutions
   - registrations
   - preprints


## QA Notes

Visting /v2/users or  /v2/nodes in list or detail view should give counts of how many items are included when you include the query parameter `related_counts=True` for specific counts use the param `related_counts=preprints` or `related_counts=preprints,nodes` etc. 

## Documentation

`related_counts` is to be documented by this: https://github.com/CenterForOpenScience/developer.osf.io/pull/21

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-846
